### ISSUE TITLE
Starting a paused partition fails

### ISSUE THREAD

**Issue by scheuran** [zhmcclient/python-zhmcclient#228]
### Actual behavior
This is the expected behavior by DPM (confirmed that) - but it's really strange. For nova-dpm we need ot fix it - the question is, if we fix it in zhmcclient or in nova-dpm.

- do an inband shutdown of a partition (e.g. via linux shell:  shutdown -P now)
- partition state switches "Paused"
- try to start that partition using "Daily - Start"

This results in the following error on the UI
"STS00005E
One or more targeted partitions are already started. Only partitions in the Stopped state can be started.
To restart partitions, first run the Stop task, and then try the operation again."

The cli does not care about this issue at all, it just leaves the partition state to paused:
$ zhmc partition list P000S67B
/+-------------------------------------------------------------+------------+
| name                                                        | status     |
|-------------------------------------------------------------+------------|
| andreas_cloud_init_tests                                    | paused     |

$ zhmc partition start P000S67B andreas_cloud_init_tests
|Partition andreas_cloud_init_tests has been started.

$ zhmc partition list P000S67B
/+-------------------------------------------------------------+------------+
| name                                                        | status     |
|-------------------------------------------------------------+------------|
| andreas_cloud_init_tests                                    | paused     |

The python binding seems to behave like the cli. It returns immediately even if wait_for_completion ==True. There's no indication that the operation failed.

### Execution environment

* zhmcclient version: 0.9.0
* Operating system (type+version):

**Comment by scheuran** [comment]
The workaround is to "stop" the partition and "start" it again

**Comment by scheuran** [comment]
This is the corresponding nova-dpm bug: https://bugs.launchpad.net/nova-dpm/+bug/1678002

**Comment by andy-maier** [comment]
Tracing the HMC interactions with the new log feature of the zhmc CLI (PR #223) reveals:

```
$ zhmc --log hmc=debug partition start P000S67B andreas_cloud_init_tests

DEBUG zhmcclient.hmc: HMC request: POST https://9.152.151.49:6794/api/partitions/eb369bcc-d97c-11e6-bd0f-42f2e9ef1641/operations/start,
    headers: {'X-API-Session': u'45cyhweet1gqg91oko7toqg64s8wu5puxvpap5lwtxs9drp7tg', 'Content-type': 'application/json', 'Accept': '*/*'},
    content(max.1000): None

DEBUG zhmcclient.hmc: HMC response: POST https://9.152.151.49:6794/api/partitions/eb369bcc-d97c-11e6-bd0f-42f2e9ef1641/operations/start,
    status: 202,
    headers: {'Content-Length': '60', 'x-request-id': 'Sx1dbdf8d8-1474-11e7-a3f7-5ef3fcb27748.37 Rx2', 'Server': 'zSeries management console API web server / 2.0', 'x-wsa-provider-duration-ms': '7', 'Location': '/api/jobs/30edc820-15f0-11e7-9470-5ef3fcb27748', 'Cache-control': 'no-cache', 'Date': 'Fri, 31 Mar 2017 08:58:21 GMT', 'Content-Type': 'application/json;charset=UTF-8'},
    content(max.1000): '{"job-uri":"/api/jobs/30edc820-15f0-11e7-9470-5ef3fcb27748"}'

DEBUG zhmcclient.hmc: HMC request: GET https://9.152.151.49:6794/api/jobs/30edc820-15f0-11e7-9470-5ef3fcb27748,
    headers: {'X-API-Session': u'45cyhweet1gqg91oko7toqg64s8wu5puxvpap5lwtxs9drp7tg', 'Content-type': 'application/json', 'Accept': '*/*'},
    content(max.1000): None

DEBUG zhmcclient.hmc: HMC response: GET https://9.152.151.49:6794/api/jobs/30edc820-15f0-11e7-9470-5ef3fcb27748,
    status: 200,
    headers: {'Content-Length': '152', 'x-request-id': 'Sx1dbdf8d8-1474-11e7-a3f7-5ef3fcb27748.37 Rx3', 'Server': 'zSeries management console API web server / 2.0', 'x-wsa-provider-duration-ms': '0', 'Cache-control': 'no-cache', 'Date': 'Fri, 31 Mar 2017 08:58:21 GMT', 'Content-Type': 'application/json;charset=UTF-8'},
    content(max.1000): '{"status":"complete","job-status-code":409,"job-results":{"error":"The partition is not in a valid state to perform the operation"},"job-reason-code":1}'

DEBUG zhmcclient.hmc: HMC request: DELETE https://9.152.151.49:6794/api/jobs/30edc820-15f0-11e7-9470-5ef3fcb27748,
    headers: {'X-API-Session': u'45cyhweet1gqg91oko7toqg64s8wu5puxvpap5lwtxs9drp7tg', 'Content-type': 'application/json', 'Accept': '*/*'},
    content(max.1000): None

DEBUG zhmcclient.hmc: HMC response: DELETE https://9.152.151.49:6794/api/jobs/30edc820-15f0-11e7-9470-5ef3fcb27748,
    status: 204,
    headers: {'Date': 'Fri, 31 Mar 2017 08:58:21 GMT', 'x-request-id': 'Sx1dbdf8d8-1474-11e7-a3f7-5ef3fcb27748.37 Rx4', 'Server': 'zSeries management console API web server / 2.0', 'x-wsa-provider-duration-ms': '0', 'Cache-control': 'no-cache'},
    content(max.1000): ''
Error: HTTPError: 409,1: None [None None]
```

This means the zhmcclient is not correctly reporting the error in the job result. This is a bug in the zhmcclient.

**Comment by andy-maier** [comment]
The zhmcclient behavior was addressed in PR #229.
